### PR TITLE
creating in-mem layer for multi-tenant schemas

### PR DESCRIPTION
**Pull request by nate-droid**
This PR will create an in-mem layer for storing graphql schemas, triggers, and schemaGraphs to support Multi-tenant services. 
Several structs needed to change to have Exported values in order to support json/byte marshalling.

### CODE REVIEW

File: store/datatree.go [hunk 1]
```diff
@@ -29,7 +29,7 @@ func (t dataTree) traverse(bCtx *env.BubblyContext, fn visitFn) (core.DataBlocks
 }
 
 // visitNode is a wrapper around the callback visit function to make sure that
-// all the nodes are visited at least once, and at most once
+// all the Nodes are visited at least once, and at most once
```
Comment (jlarfors) [inline store/datatree.go:32]: Have you done a search+replace here and changed these by accident, or is there a reason for all these being capitalised now?

Comment (nate-droid) [reply store/datatree.go:32]: I had to export the fields, otherwise json marshalling wouldn't work

Comment (nate-droid) [reply store/datatree.go:32]: (and this part must have gotten changed automatically with my refactor)

File: store/mem.go [hunk 2]
```diff
@@ -0,0 +1,136 @@
...
+	if updateErr != nil {
+		return updateErr
+	}
+	return nil
+}
+
+func fetchElement(tenantID string, bucketID string) ([]byte, error) {
+	db, err := bolt.Open(dbPath, 0666, nil)
+	if err != nil {
+		return nil, err
```
Comment (jlarfors) [inline store/mem.go:47]: Should there be some context added to the err message? `fmt.Errorf("failed to open bolt db: %w", err)` ?

Comment (jlarfors) [reply store/mem.go:47]: Seems like just `err` is being returned everywhere. Perhaps it's sufficient

File: store/datatree.go (hunk 1 above)
Comment (olliefr) [reply store/datatree.go:32]: Another approach would be to restrict the scope of change to marshalling methods by using anonymous structs:

[Marshaling structs with private fields](https://riptutorial.com/go/example/14194/marshaling-structs-with-private-fields)

Just an idea, depending on how much do we really want the "node" fields to be private.

Comment (jlarfors) [reply store/datatree.go:32]: Neat trick, but I guess we can keep the members public for now.

**Review by olliefr — APPROVED** [review]
With the benefit of hindsight, I'd do this as two separate PRs:

1. To expose `node` fields as public;
2. To add in-memory layer for multi-tenant schemas.

Otherwise, LGTM ✔️

**Review by jlarfors — APPROVED** [review]

**Pull request merged**